pkg/cloudprovider/aws/instancetype.go

```diff
@@ -46,13 +46,21 @@ type InstanceType struct {
 	resources    v1.ResourceList
 	provider     *v1alpha1.AWS
 	maxPods      *int32
+	price        float64
 }
 
-func NewInstanceType(ctx context.Context, info *ec2.InstanceTypeInfo, provider *v1alpha1.AWS, offerings []cloudprovider.Offering) *InstanceType {
+func NewInstanceType(ctx context.Context, info *ec2.InstanceTypeInfo, pricing *PricingProvider, provider *v1alpha1.AWS, offerings []cloudprovider.Offering) (*InstanceType, error) {
+	// TODO: move pricing information from the instance type down into offerings
```

Comment:
Leaving this for a future PR.  Our pricing information is at the wrong level at the moment (instance vs offering), it also varies by OS.